common/net/http/withBody_test.go

```diff
@@ -42,3 +44,35 @@ func TestNewOfTypeWithComplexStruct(t *testing.T) {
 		t.Error("Wrong data.")
 	}
 }
+
+func TestFilterRequiredFields(t *testing.T) {
```

Comment:
Use the table convention for testing in this case

`
func TestFieldsRequired(t *testing.T) {
	tests := []struct {
		name     string
		input    common.FieldValidations
		expected common.FieldValidations
	}{
		{
			name:     "Nil Map",
			input:    nil,
			expected: common.FieldValidations{},
		},
		{
			name: "Empty Map",
			input: common.FieldValidations{},
			expected: common.FieldValidations{},
		},
		{
			name: "No Required Fields",
			input: common.FieldValidations{
				"field1": "optional",
				"field2": "not required",
			},
			expected: common.FieldValidations{},
		},
		{
			name: "Some Required Fields",
			input: common.FieldValidations{
				"field1": "required",
				"field2": "optional",
				"field3": "Required for processing",
			},
			expected: common.FieldValidations{
				"field1": "required",
				"field3": "Required for processing",
			},
		},
		{
			name: "Mixed Case Required Fields",
			input: common.FieldValidations{
				"field1": "REQUIRED",
				"field2": "Optional",
				"field3": "reQuIrEd for use",
			},
			expected: common.FieldValidations{
				"field1": "REQUIRED",
				"field3": "reQuIrEd for use",
			},
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := yourmodule.FieldsRequired(tt.input)

			// Verifica se o resultado é igual ao esperado
			if len(result) != len(tt.expected) {
				t.Errorf("expected length %d, got %d", len(tt.expected), len(result))
			}

			for k, v := range tt.expected {
				if result[k] != v {
					t.Errorf("expected %s: %s, got %s", k, v, result[k])
				}
			}
		})
	}
}
`